File: Management/Models/CanvasModels/Assignments/CanvasRubricCriteria.cs
namespace CanvasModel.Assignments;

public record CanvasRubricCriteria
(
  [property: JsonPropertyName("id")]
  string Id,

  [property: JsonPropertyName("description")]
  string Description,

  [property: JsonPropertyName("long_description")]
  string LongDescription,

  [property: JsonPropertyName("points")]
  double? Points,
  [property: JsonPropertyName("learning_outcome_id")]
  string? LearningOutcomeId,

  [property: JsonPropertyName("vendor_guid")]
  string? VendorGuid,

  [property: JsonPropertyName("criterion_use_range")]
  bool? CriterionUseRange = null,

  [property: JsonPropertyName("ratings")]
  IEnumerable<CanvasRubricRating>? Ratings = null,

  [property: JsonPropertyName("ignore_for_scoring")]
  bool? IgnoreForScoring = null
);

File: src/models/canvas/assignments/canvasRubricCriteria.ts
export interface CanvasRubricCriteria {
  id: string;
  description: string;
  long_description: string;
  points?: number;
  learning_outcome_id?: string;
  vendor_guid?: string;
  criterion_use_range?: boolean;
  ratings?: {
    points: number;
    id: string;
    description: string;
    long_description: string;
  }[];
  ignore_for_scoring?: boolean;
}
